### PR TITLE
refactored create function

### PR DESCRIPTION
**Pull request by MrishoLukamba**
This PR refactors `create function `in pallet-asset, the refactoring function is `do_create` .
   And this solves issue #11359

### CODE REVIEW

**Comment by MrishoLukamba** [comment]
Done formatting it

File: frame/assets/src/functions.rs [hunk 1]
```diff
@@ -620,6 +620,55 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
...
+	/// Parameters:
+	/// - `id`: The identifier of the new asset. This must not be currently in use to identify
+	/// an existing asset.
+	/// - `admin`: The admin of this class of assets. The admin is the initial address of each
+	/// member of the asset class's admin team.
+	/// - `min_balance`: The minimum balance of this new asset that any single account must
+	/// have. If an account's balance is reduced below this, then it collapses to zero.
+	///
+	/// Emits `Created` event when successful.
+	pub(super) fn do_create(
```
Comment (shawntabrizi) [inline frame/assets/src/functions.rs:639]: pub(super) does not make sense if you are trying to expose these functions to external crates. Should just be `pub`

Comment (KiChjang) [reply frame/assets/src/functions.rs:639]: Yes; in fact, most of the functions in this file needs to be changed to `pub` when the intention here is to expose them as a public API.

File: frame/assets/src/functions.rs [hunk 2]
```diff
@@ -812,6 +861,192 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
...
 	}
 
+	/// Disallow further unprivileged transfers for the asset class.
+	///
+	/// Origin must be Signed and the sender should be the Freezer of the asset `id`.
+	///
+	/// - `id`: The identifier of the asset to be frozen.
+	///
+	/// Emits `Frozen`.
+	pub(super) fn do_freeze_asset(origin: T::AccountId, id: T::AssetId) -> DispatchResult {
```
Comment (shawntabrizi) [inline frame/assets/src/functions.rs:871]: would not use "origin" here. maybe "sender" or "who"

File: frame/assets/src/functions.rs [hunk 3]
```diff
@@ -812,6 +861,192 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		Ok(())
 	}
 
+	/// Disallow further unprivileged transfers for the asset class.
+	///
+	/// Origin must be Signed and the sender should be the Freezer of the asset `id`.
+	///
+	/// - `id`: The identifier of the asset to be frozen.
+	///
```
Comment (shawntabrizi) [inline frame/assets/src/functions.rs:869]: If you are going to comment the parameters, you should at least include all of them.

File: frame/assets/src/functions.rs [hunk 4]
```diff
@@ -812,6 +861,192 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
...
+			ensure!(origin == d.freezer, Error::<T, I>::NoPermission);
+
+			d.is_frozen = true;
+
+			Self::deposit_event(Event::<T, I>::AssetFrozen { asset_id: id });
+			Ok(())
+		})
+	}
+	/// This fn do_tansfer_ownership
+	pub(super) fn do_tansfer_ownership(
```
Comment (shawntabrizi) [inline frame/assets/src/functions.rs:883]: typos and an awful comment.

File: frame/assets/src/functions.rs [hunk 5]
```diff
@@ -812,6 +861,192 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
...
+			let d = maybe_details.as_mut().ok_or(Error::<T, I>::Unknown)?;
+			ensure!(origin == d.admin, Error::<T, I>::NoPermission);
+
+			d.is_frozen = false;
+
+			Self::deposit_event(Event::<T, I>::AssetThawed { asset_id: id });
+			Ok(())
+		})
+	}
+	/// This fn thaw
```
Comment (shawntabrizi) [inline frame/assets/src/functions.rs:926]: I would not accept a PR with this quality of code comment

**Comment by shawntabrizi** [comment]
Generally this PR does not make sense.

The correct way to do this would be to create traits which expose these functions, and afaik, a lot of these functions / traits already exist. That is what this entire file is:

https://github.com/paritytech/substrate/blob/master/frame/assets/src/impl_fungibles.rs

And all the traits here:

https://github.com/paritytech/substrate/tree/master/frame/support/src/traits/tokens

**Comment by KiChjang** [comment]
No activity from the PR author, closing. Feel free to reopen if you want to continue working on it.

**Pull request closed without merging**